node/core/dispute-coordinator/src/real/spam_slots.rs

```diff
@@ -90,14 +98,16 @@ impl SpamSlots {
 		let validators = self.unconfirmed.entry((session, candidate)).or_default();
 
 		if validators.insert(validator) {
+			// We only increment spam slots once per candidate, as each validator has to provide an
+			// opposing vote for sending out its own vote. Therefore, receiving multiple votes for
+			// a single candidate is expected and should not get punished here.
```

Comment:
Testing my understanding - a specific 'Invalid' vote might get imported as a part of several `DisputeMessages` and after the first import this would previously return `false`, which is meant to indicate that spam slots are full. Is that right?

Reply:
yes! You already made me fix that back then, but I "fixed" it wrongly unfortunately :1st_place_medal: